includes/Core/Consent_Mode/REST_Consent_Mode_Controller.php

```diff
@@ -163,13 +164,19 @@ protected function get_rest_routes() {
 								if ( ! function_exists( 'get_plugins' ) ) {
 									require_once ABSPATH . 'wp-admin/includes/plugin.php';
 								}
-								foreach ( array_keys( get_plugins() ) as $installed_plugin ) {
-									if ( $installed_plugin === $plugin ) {
+								foreach ( get_plugins() as $plugin_file => $installed_plugin ) {
+									if ( $installed_plugin['PluginURI'] === $plugin_uri ) {
+										$slug      = $installed_plugin['TextDomain'];
```

Comment:
The `slug` is only used for the `$install_url` so this value shouldn't be dynamic based on the match as we should only install the specific/known plugin.

Basically, we should only be detecting whether the plugin is installed here or not, and the plugin basename which is needed for constructing the `$activate_url`.